Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUShaderModuleCompilationHint.h

```diff
@@ -25,12 +25,16 @@
 
 #pragma once
 
-namespace WebCore::WebGPU {
+#include "WebGPUPipelineLayout.h"
+#include <wtf/Ref.h>
+#include <wtf/WeakRef.h>
 
-class WebGPUPipelineLayout;
+namespace WebCore::WebGPU {
 
 struct ShaderModuleCompilationHint {
-    PipelineLayout& pipelineLayout;
+    WeakRef<PipelineLayout> pipelineLayout;
+
+    Ref<const PipelineLayout> protectedPipelineLayout() const { return pipelineLayout.get(); }
```

Comment:
You're using `WeakRef<PipelineLayout>` above, not `WeakRef<const PipelineLayout>` so not sure why we're adding extra constness here.

Reply:
oh, I thought I had to match constness but I guess not. Will change.